### PR TITLE
feat: activate redirect experiment to split traffic between lend filt…

### PR DESCRIPTION
**Pull request by mcstover**
…er pages

### CODE REVIEW

File: src/graphql/query/lendFilterPage.graphql [hunk 1]
```diff
@@ -15,10 +15,4 @@ query lendFilterPage($basketId: String) {
 			}
 		}
 	}
-	general {
```
Comment (mcstover) [inline src/graphql/query/lendFilterPage.graphql:18]: EXP setting fetch has moved to new async function in the preFetch promise chain

File: src/pages/Lend/Filter/LendFilterPage.vue [hunk 2]
```diff
@@ -134,6 +137,30 @@ import lendFilterPageQuery from '@/graphql/query/lendFilterPage.graphql';
...
+	const eligibleQueryParams = ['page', 'sortBy', 'gender', 'registration'];
+	const queryParamKeys = Object.keys(route?.query);
+	const allowedSorts = ['expiringSoon', 'loanAmountDesc', 'loanAmount', 'popularity'];
+	// eligible by default, no params is also eligible
+	let isEligible = true;
+
+	queryParamKeys.forEach(queryParam => {
+		// ensure any query params are eligible
+		if (!eligibleQueryParams.includes(queryParam)) {
+			isEligible = false;
```
Comment (dyersituations) [inline src/pages/Lend/Filter/LendFilterPage.vue:153]: Not a huge performance difference in this situation, but you could use a short circuited loop.

File: src/pages/Lend/Filter/LendFilterPage.vue [hunk 3]
```diff
@@ -182,23 +209,18 @@ export default {
...
-
-		if (lendFilterFlssExp.version && lendFilterFlssExp.version !== 'unassigned') {
-			if (lendFilterFlssExp.version === 'b') {
-				this.hideNonFlssFilters = true;
-				this.initiallyExpandedFilters = true;
+		if (isFLSSEligible(this.$route)) {
+			const { enabled } = getExperimentSettingCached(this.apollo, lendFilterRedirectEXP);
+			if (enabled) {
+				// this method will get the version from the apollo cache
+				trackExperimentVersion(
```
Comment (dyersituations) [inline src/pages/Lend/Filter/LendFilterPage.vue:216]: Is there a reason we track this here instead of on the FLSS filter page?

Comment (mcstover) [reply src/pages/Lend/Filter/LendFilterPage.vue:216]: This event will fire for visitors in the control group of the experiment with `a` as the "version" whereas the event on our new page will fire with `b` as the version. This page will redirect before it's loaded in the browser if the are in the experiment group and the event fires in the client.

File: src/pages/Lend/Filter/LendFilterPage.vue [hunk 4]
```diff
@@ -134,6 +137,30 @@ import lendFilterPageQuery from '@/graphql/query/lendFilterPage.graphql';
 
 import lendFilterExpMixin from '@/plugins/lend-filter-page-exp-mixin';
 
+const lendFilterRedirectEXP = 'lend_filter_flss_v1';
+
+function isFLSSEligible(route = {}) {
+	// check route for eligibility
+	const eligibleQueryParams = ['page', 'sortBy', 'gender', 'registration'];
```
[the remark this answers is not in the document]
Comment (dyersituations) [reply src/pages/Lend/Filter/LendFilterPage.vue:144]: Thanks for the info. We already are mapping the Algolia names to IDs for `sector` and `attribute`.

File: src/pages/Lend/Filter/LendFilterPage.vue (hunk 3 above)
Comment (dyersituations) [reply src/pages/Lend/Filter/LendFilterPage.vue:216]: Ah, so we're tracking eligibility regardless.

**Review by dyersituations — APPROVED** [review]

File: src/pages/Lend/Filter/LendFilterPage.vue [hunk 5]
```diff
@@ -134,6 +137,30 @@ import lendFilterPageQuery from '@/graphql/query/lendFilterPage.graphql';
 
 import lendFilterExpMixin from '@/plugins/lend-filter-page-exp-mixin';
 
+const lendFilterRedirectEXP = 'lend_filter_flss_v1';
```
[the remark this answers is not in the document]
Comment (mcstover) [reply src/pages/Lend/Filter/LendFilterPage.vue:140]: We'll walk through this setup together soon.

**Pull request merged**